### ISSUE TITLE
Deprecate using ansible-container to build images

### ISSUE THREAD

**Issue by rthallisey** [fusor/ansible-playbook-bundle-library#2]
Right now we have two ways to build images: ansible-container and docker. This forces the APB maintainers to support a second set of resources for an image in the ```ansible``` directory.

I think we should move away from ansible-container and use regular docker build. We have already built up a set of Dockerfiles that can be used as examples for building new ones.  This will make it less confusing for devs building their APBs and easier for the maintainers to keep these containers working.

**Comment by fabianvf** [comment]
As far as I know we don't require you to use ansible-container, we just allow it. You should be able to just make a Dockerfile based apb with no ansible-container support without issue.

**Comment by rthallisey** [comment]
I agree, you can make an apb without ansible-container.  But, I think we should remove all the directories and playbooks the are used with ansible-container and stick to one build method: ```docker build```.

**Comment by fabianvf** [comment]
The docs for [isvs](https://github.com/fusor/ansible-playbook-bundle/blob/master/docs/isv.md) and [developers](https://github.com/fusor/ansible-playbook-bundle/blob/master/docs/developers.md) also currently includes the ansible-container workflow, so if we're planning on removing it we should probably start with a proposal there. 

AFAIK collaboration with ansible-container to allow users to define their apbs with ansible was a goal from the start, and a differentiator for our broker/format that would help with buy-in and integration from isvs and developers. I'm not sure what we gain by dropping support for it.

**Comment by rthallisey** [comment]
If isvs and devs want to use ansible-container they can.  What I'm proposing is we remove the directory structure for ansible-container and only provide Dockerfiles, scripts, and playbooks for each container.  The support for ansible-container doesn't change, except the dev would have to do ```ansible-container init``` to setup their build directory and move the playbooks into place.

There's also tons of other build tools and CI/CD pipelines out there and our repo should not favor any of them.  
The main focus of the library should be make sure standard the docker commands work and nothing else.  If we did that I think it would simplify things and make them easier to consume.

**Comment by fabianvf** [comment]
@jwmatthews @jmrodri @eriknelson @dymurray want to weigh in?

**Comment by jmrodri** [comment]
@rthallisey so my understanding of what you're saying is we keep our `apb` directory and standard `Dockerfile`s and 'REMOVE' the `ansible` directory. If a user wants to use `ansible-container` to start they can. The files that are generated they can copy them into the appropriate `apb` directory. 

I am not opposed to that idea. I do think our docs should still document how to use `ansible-container` in the workflow as an example of how we integrate with other tools. And definitely how to create with just a straight up `Dockerfile`.

**Comment by jwmatthews** [comment]
@rthallisey @fabianvf @jmrodri let's discuss with the team after Scrum on Tuesday.  I'll block out time on our calendars.

**Comment by tsanders-rh** [comment]
Keep in mind that our goal is to provide dev tooling for this process.  We talk about 3 steps: (1) prepare, (2) package, (3) deploy.  We should just have consistent tooling for each of the 3 steps.  ansible-container should just be used like a black-box tool / library.  For the guided workflow, our users shouldn’t have to know anything about ansible-container or interact with their CLI, etc…..our prepare and package steps should do this for them.

-Todd

**Comment by rthallisey** [comment]
@jmrodri exactly, we remove the ansible directory. We can still document how to use ansible-container or any other tool for that matter when creating an apb.

**Comment by rthallisey** [comment]
After further discussion, the consensus is to move the ansible directory into the apb directory.  That way the apb directory has the structure of an ansible playbook and we're not carrying an extra directory that is only used by a specific tool.
